sftp/sftp.go

```diff
@@ -455,9 +498,37 @@ func (o *Object) Remote() string {
 	return o.remote
 }
 
-// Hash returns "" since SFTP (in Go or OpenSSH) doesn't support remote calculation of hashes
 func (o *Object) Hash(r fs.HashType) (string, error) {
-	return "", fs.ErrHashUnsupported
+	session, err := o.fs.sshClient.NewSession()
```

Comment:
I can see why, but this might be difficult to implement. Simply caching the hash for every object would be inefficient. Even just considering the RAM usage (think systems like a Raspberry Pi): SHA1 is 160 bits, so for e.g. 10 million files that would be ~ 200 MB just for the hashes, not even considering file path strings.

So, the way to go would be probably be either an LRU cache or a limited priority-cache with file sizes as priorities, since large files are expensive to hash. Since I'm unfamiliar with the Go world: Do you know of a good and easy-to-use library for that? If not and you'd really like to see caching implemented here, I could do more research on this.